Danki2/Assets/Scripts/Actor/Mortal/Player/Player.cs

```diff
@@ -1,12 +1,80 @@
-﻿using System.Collections;
+﻿using System;
+using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
 
+public enum CastingStatus
+{
+    ChannelingLeft,
+    ChannelingRight,
+    Cooldown,
+    Ready
+}
+
+public enum CastingCommand
+{
+    ContinueChannel,
+    CancelChannel,
+    CastLeft,
+    CastRight,
+    None
+}
+
+public enum ActionControlState
+{
+    Left,
+    Right,
+    Both,
+    None
+}
+
 public class Player : Mortal
 {
+    private AbilityTree _abilityTree;
+    private ChannelService _channelService;
+    private CastingStatus _castingStatus = CastingStatus.Ready;
+    private ActionControlState _previousActionControlState = ActionControlState.None;
+    private float _remainingCooldown = 0f;
+    private readonly float _maximumCooldown = 1f;
+
+    public override void Start()
+    {
+        base.Start();
+
+        _channelService = new ChannelService();
+
+        _abilityTree = AbilityTreeFactory.CreateTree(
+            AbilityTreeFactory.CreateNode(
+                AbilityReference.Slash,
+                AbilityTreeFactory.CreateNode(AbilityReference.ShieldBash),
+                AbilityTreeFactory.CreateNode(AbilityReference.Whirlwind)
+            ),
+            AbilityTreeFactory.CreateNode(
+                AbilityReference.Whirlwind,
+                AbilityTreeFactory.CreateNode(AbilityReference.Slash),
+                AbilityTreeFactory.CreateNode(AbilityReference.ShieldBash)
+            )
+        );
+    }
+
+    public override void Update()
+    {
+        _remainingCooldown = Mathf.Max(0f, _remainingCooldown - Time.deltaTime);
+        if (_remainingCooldown == 0f && _castingStatus == CastingStatus.Cooldown)
+        {
+            _castingStatus = CastingStatus.Ready;
+        }
+
+        _channelService.Update();
+
+        base.Update();
```

Comment:
Does `base.Update()` get called at the end so that `_statsManager.Rebase();` and `_effectTracker.ProcessEffects();` get called at the end of the update? I think that's the right thing to do. There's a LateUpdate() function that gets called after all of the Update() functions, we could move those function calls above into there in Actor. Then `base.Update()` can get called at the start of this method (I think that's normally best practice when overriding)

Reply:
I'm thinking we should just call base.Update first. Stats and effects can certainly be sorted out at the start of the frame for me.